virtual/registry/kv_registry.go

```diff
@@ -431,15 +434,28 @@ func (k *kvRegistry) Heartbeat(
 
 		var state serverState
 		if !ok {
+			serverVersion = 0
 			state = serverState{
-				ServerID: serverID,
+				ServerID:      serverID,
+				ServerVersion: serverVersion,
 			}
+			state.LastHeartbeatedAt = time.Now()
 		} else {
 			if err := json.Unmarshal(v, &state); err != nil {
 				return nil, err
 			}
 		}
 
+		var (
+			timeSinceLastHeartbeat = time.Since(state.LastHeartbeatedAt)
```

Comment:
super nit: `timeSinceLastHeartbeat :=` I only use `var` blocks for more than one declaration